interceptor/pkg/interceptor/pdinterceptor_test.go

```
@@ -223,14 +227,140 @@ func TestOversizedRequestBodyIsRejected(t *testing.T) {
+	sig := pdSignatureFor(signingSecret, innerBody)
+
+	outer, err := json.Marshal(map[string]interface{}{
+		"body": innerBody,
+		"header": map[string][]string{
+			"X-PagerDuty-Signature": {sig},
+		},
+	})
+	if err != nil {
+		t.Fatalf("makeSignedRequest: marshal: %v", err)
+	}
+	return httptest.NewRequest(http.MethodPost, "/", bytes.NewReader(outer))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)